apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -75,19 +79,44 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
 
         HttpServletRequest request = (HttpServletRequest)ContextManager.getRuntimeContext().get(REQUEST_KEY_IN_RUNTIME_CONTEXT);
         if (request != null) {
-            ContextCarrier contextCarrier = new ContextCarrier();
-            CarrierItem next = contextCarrier.items();
-            while (next.hasNext()) {
-                next = next.next();
-                next.setHeadValue(request.getHeader(next.getHeadKey()));
+            Object stackDepth = ContextManager.getRuntimeContext().get(CONTROLLER_METHOD_STACK_DEPTH);
+            Integer depth = stackDepth == null ? 0 : Integer.parseInt(stackDepth.toString());
```

Comment:
Such as here, should be 

Also add a static method in `ControllerDepth` class, to 
1. Process not exist 
1. Get from RuntimeContext, always return `ControllerDepth` instance.
1. After ControllerDepth becomes an object, you don't need to put it many times, because it is a pointer.